CtrlVAF/CtrlVAF/Events/EventDispatcher.cs

```diff
@@ -87,18 +93,19 @@ protected internal override IEnumerable<Type> GetTypes(params ICtrlVAFCommand[]
                         t.BaseType.GetGenericTypeDefinition() == abstractHandlerType &&
                         t.BaseType.GenericTypeArguments[1] == commandType); ;
 
-                    var eventHanderTypes = commandHandlerTypes.Where(t =>
+                    allCommandHandlerTypes.AddRange(commandHandlerTypes);
+                }
+
+                //Was there a reason for this Distinct? The same assembly is not included twice.
+                TypeCache.TryAdd(commandType, allCommandHandlerTypes.Distinct());
```

Comment:
I suppose there is not a particular reason to have distinct here.
```suggestion
                TypeCache.TryAdd(commandType, allCommandHandlerTypes);
```